src/PaySimpleSdk/Accounts/Validation/AccountValidator.cs

```diff
@@ -50,10 +50,78 @@ public AchValidator()
     internal class CreditCardValidator : AccountValidator<CreditCard>
     {
         public CreditCardValidator()
-        {
-            RuleFor(m => m.CreditCardNumber).NotEmpty().WithMessage("CreditCardNumber is required").Matches(@"^(?:4[0-9]{12}(?:[0-9]{3})?|5[1-5][0-9]{14}|3[47][0-9]{13}|6(?:011|5[0-9]{2})[0-9]{12})$|^((\*{11,12})[0-9]{4})$").WithMessage("CreditCardNumber is invalid");
-            RuleFor(m => m.ExpirationDate).NotEmpty().WithMessage("ExpirationDate is required").Matches(@"^(([0][1-9])|([1][0-2]))/20[0-9][0-9]$").WithMessage("ExpirationDate must be in a \"MM/YYYY\" format");
+		{
+			RuleFor(m => m.CreditCardNumber)
+				.NotEmpty()
+				.WithMessage("CreditCardNumber is required");
+
+			RuleFor(m => m.CreditCardNumber)
+				.Must(a => string.IsNullOrWhiteSpace(a) || a.Contains("*") || LuhnAlgorithm.IsValidCreditCard(a))
+				.WithMessage("CreditCardNumber is invalid");
+
+			RuleFor(m => m.CreditCardNumber)
+				.Matches(@"^(?:4[0-9]{12}(?:[0-9]{3})?|5[1-5][0-9]{14}|3[47][0-9]{13}|6(?:011|5[0-9]{2})[0-9]{12})$|^((\*{11,12})[0-9]{4})$")
```

Comment:
This regex won't work for 19 digit credit cards which we are now supporting

Reply:
Yeah i replaced the regex with a Luhn validator for cc cards and i just added a test to verify it works.

The regex I left things in for numbers with *'s only. I have no idea why children's miracle network added validations that allow *'s, but I didn't want to break their code or the tests so I left it in that use case.